solr/core/src/test/org/apache/solr/update/processor/DistributedUpdateProcessorTest.java

```diff
@@ -220,4 +185,45 @@ protected boolean tryLock(int lockTimeoutMs) {
       return succeeded;
     }
   }
+
+  private static DistributedUpdateProcessor newDurp(SolrQueryRequest req, long lockTimeoutMs) {
+    if (lockTimeoutMs <= 0) {
+      // default
+      return new DistributedUpdateProcessor(req, null, null);
+    }
+    // customize UpdateLocks with the provided timeout.  And simulate docs taking longer to index
+    final var sleepMs = 5000;
```

Comment:
Yeah I was thinking that (this would be an improvement over the test as it ws written).  I'll do that; maybe still giving it more time in case the test is running slow.